iscservice/demo/libdemo.h

```diff
@@ -0,0 +1,71 @@
+/* -*- C -*- */
+/*
+ * Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ */
+
+#pragma once
+
+#ifndef __MOTR_ISCSERVICE_DEMO_LIBDEMO_H__
+#define __MOTR_ISCSERVICE_DEMO_LIBDEMO_H__
+
+#include "lib/types.h"
+#include "lib/buf.h"
+#include "lib/buf_xc.h"
+#include "fid/fid.h"
+#include "fid/fid_xc.h"
+#include "lib/vec.h"
+#include "lib/vec_xc.h"
+#include "xcode/xcode_attr.h"
+
+/**
+ * Holds the result of min and max computations.
+ *
+ * The left and right cuts of the values which cross
+ * the units boundaries should be glued by the client code
+ * and included in the final computation also.
+ */
+struct mm_result {
+	/** Index of the resulting element. */
+	uint64_t      mr_idx;
+	/** Total number of elements. */
+	uint64_t      mr_nr;
+	/** The resulting value of the computation. */
+	double        mr_val;
+	/** Right cut of the boundary value on the left side of unit. */
+	struct m0_buf mr_lbuf;
+	/** Left cut of the boundary value on the right side of unit. */
+	struct m0_buf mr_rbuf;
+} M0_XCA_RECORD;
+
+/** Arguments to the target ISC service. */
+struct isc_targs {
+	struct m0_fid         ist_cob;
+	struct m0_io_indexvec ist_ioiv;
```

Comment:
Only "len" is used in this demo (offset is supposed to be 0, from start).
No need to use "m0_io_indexvec".
I know, "m0_io_indexvec" can be more general. But, as I said, we just need the sample demo app to be as simple as possible.

Reply:
Would this argument be generic to all ISC?

Reply:
> Only "len" is used in this demo (offset is supposed to be 0, from start).

Why? This is the location of the unit in the cob, so the offset and the length are used for each request.

> Would this argument be generic to all ISC?

I don't think this will be changed too much, yes. But the user may put anything there that is needed for the computation at the server side.

Reply:
I misunderstood this originally.
After more reading, I understand this is the io vectors for a particular cob.
This is generated by layout plan.

Reply:
Added a few comments into the structure to make it more clear (hopefully).